frontend/scripts/react-components/dashboard-element/dashboard-widget/table-modal/table-modal.selectors.js

```diff
@@ -0,0 +1,149 @@
+import { createSelector } from 'reselect';
+import isEmpty from 'lodash/isEmpty';
+import { format } from 'd3-format';
+import { CHART_TYPES } from 'constants';
+
+const getMeta = (state, { meta }) => meta || null;
+const getData = (state, { data }) => data || null;
+const getChartType = (state, { chartType }) => chartType || null;
+const returnNameArrayIfNotEmpty = value =>
+  isEmpty(value) ? null : Object.values(value).map(v => v.name);
+const getCountriesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.countriesPanel.activeItems
+  );
+const getCommoditiesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.commoditiesPanel.activeItems
+  );
+const getSourcesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.commoditiesPanel.activeItems
+  );
+
+const getVariableColumnName = createSelector(
+  [getMeta],
+  meta => (meta.info.node_type === 'COUNTRY' ? 'DESTINATION' : meta.info.node_type)
+);
+
+const hasNContIndicator = createSelector(
+  [getMeta],
+  meta => meta.info.filter.ncont_attribute
+);
+const hasMultipleYears = createSelector(
+  [getMeta],
+  meta => meta.info.years.end_year
+);
+const hasVariableColumn = createSelector(
+  [getChartType, getVariableColumnName, hasMultipleYears, hasNContIndicator, getSourcesPanel],
```

Comment:
I usually call the fns as `getHasMultipleYears` that way the value doesn't need the underscore. That said, there's no need to change this.